File: 5Days-SoftwareArchitecture/Java/Day3/Complexity/bad_nested_bank.java
class BankService {

    public void withdraw(double balance, double amount, boolean otpVerified, boolean isHoliday, String accountType) {

        if (balance > 0) {                                 // 1
            if (amount <= balance) {                      // 2
                if (otpVerified) {                        // 3
                    if (!isHoliday) {                     // 4
                        switch (accountType) {            // 5
                            case "SAVINGS":
                                System.out.println("Withdraw Successful from Savings");
                                break;
                            case "CURRENT":
                                System.out.println("Withdraw Successful from Current");
                                break;
                            default:
                                System.out.println("Invalid Account");
                                break;
                        }
                    } else {
                        System.out.println("Cannot withdraw: Bank is closed today (holiday).");
                    }
                } else {
                    System.out.println("Cannot withdraw: OTP not verified.");
                }
            } else {
                System.out.println("Cannot withdraw: Insufficient balance.");
            }
        } else {
            System.out.println("Cannot withdraw: Account balance must be positive.");
        }
    }
}

public class bad_nested_bank {
    public static void main(String[] args) {
        BankService svc = new BankService();

        System.out.println("=== Test 1: Successful withdraw (SAVINGS) ===");
        svc.withdraw(1000.0, 100.0, true, false, "SAVINGS");

        System.out.println("\n=== Test 2: Successful withdraw (CURRENT) ===");
        svc.withdraw(500.0, 200.0, true, false, "CURRENT");

    }
}

File: 5Days-SoftwareArchitecture/typesript/Day3/Complexity/bad_nested_bank.js
var BankServiceNested = /** @class */ (function () {
    function BankServiceNested() {
    }
    BankServiceNested.prototype.withdraw = function (balance, amount, otpVerified, isHoliday, accountType) {
        if (balance > 0) { // decision 1
            if (amount <= balance) { // decision 2
                if (otpVerified) { // decision 3
                    if (!isHoliday) { // decision 4
                        switch (accountType) { // decision 5 (switch)
                            case "SAVINGS": // case path 1
                                console.log("Withdraw Successful from Savings");
                                break;
                            case "CURRENT": // case path 2
                                console.log("Withdraw Successful from Current");
                                break;
                            default:
                                console.log("Invalid Account");
                                break;
                        }
                    }
                }
            }
        }
    };
    return BankServiceNested;
}());
var svc = new BankServiceNested();
svc.withdraw(1000, 100, true, false, "SAVINGS");
